nuxeo-core/nuxeo-core-cache/src/main/java/org/nuxeo/ecm/core/cache/CacheServiceImpl.java

```diff
@@ -49,102 +53,143 @@
 
     private static final Log log = LogFactory.getLog(CacheServiceImpl.class);
 
-    protected final CacheRegistry cacheRegistry = new CacheRegistry();
+    protected final CacheDescriptorRegistry registry = new CacheDescriptorRegistry();
 
-    /**
-     * Contains the names of all caches which have not been registered from an extension
-     */
-    protected final List<String> autoregisteredCacheNames = new ArrayList<>();
+    /** Currently registered caches. */
+    protected final Map<String, CacheManagement> caches = new ConcurrentHashMap<>();
 
-    @Override
-    public Cache getCache(String name) {
-        return cacheRegistry.getCache(name);
-    }
+    protected boolean started;
```

Comment:
You can probably rely on `Framework.getRuntime().getComponentManager().isStarted()`.